### PR TITLE
Add Local Scaling & Normalization Methods to Torch

### PR DESCRIPTION
**Pull request by marxprop**
Added new scaling methods to Pytorch  which include: 
- Global Normalization
- Local Normalization,
- Local Scaling and
-  MinMax Scaler

All the scaling methods except from Global Normalizing are done online for each image while training. It is applied after all data augmentation transforms, if any.  

Global Normalization is set as the default to `dgbpy.mlapply.getScaledTrainingData()` using the global mean and std of all the samples for standard scaling.

The Global Normalization method is applied to all workflows with Log output. `dgbpy.hdf5.isLogOutput()`

Information about the scaling method and its property is also saved to any trained model.

Increased the default number of Unet blocks in Pytorch from 1 to 2 so models can learn better.

Remove dict support for pytorch transforms. UI now uses default transform parameters available in `dgbpy.dgbtorch.torch_dict`.

### CODE REVIEW

File: dgbpy/dgbtorch.py [hunk 1]
```diff
@@ -20,10 +20,10 @@
 
 device = torch.device('cuda:0' if torch.cuda.is_available() else 'cpu')
 
-transform_dict = {
-    'RandomFlip':{'p':0.3},
-    'RandomGaussianNoise' :{'p':0.3, 'std':0.3}
-}
+transform_dict = [
```
Comment (waynegm) [inline dgbpy/dgbtorch.py]: transform_list

File: dgbpy/torch_classes.py [hunk 2]
```diff
@@ -714,18 +715,17 @@ def __repr__(self):
...
-    def update_pars(self, transform_dict):
-        self.p = float(transform_dict['p'])
+    def transformLabel(self, info):
+        return dgbhdf5.isImg2Img(info)
 
-    def __call__(self, image, label, image_only = False):
+    def __call__(self, image=None, label=None):
         if self.p > np.random.uniform(0,1):
-            if image_only:
+            if isinstance(label, type(None)):
```
Comment (waynegm) [inline dgbpy/torch_classes.py]: Would "if not label:" have the same result?

File: dgbpy/torch_classes.py [hunk 3]
```diff
@@ -746,82 +746,129 @@ def transform(self, arr):
...
+class ScaleTransform():
+    def __init__(self):
+        from dgbpy.dgbscikit import scale
+        self.scale = scale
+
+    def transformLabel(self, info):
+        return dgbhdf5.doOutputScaling(info)       
+
+    def __call__(self, image=None, label=None):
+        if isinstance(label, type(None)):
```
Comment (waynegm) [inline dgbpy/torch_classes.py]: Would "if not label:" have the same result?

File: dgbpy/torch_classes.py [hunk 4]
```diff
@@ -746,82 +746,129 @@ def transform(self, arr):
...
+    dgbkeys.minmaxtypestr: MinMaxScaler
 }
 
 class TransformComposefromList():
-    def __init__(self, transforms):
+    """
+        Applies all the transform from a list to the data.
+    """
+    def __init__(self, transforms, info):
         if not hasattr(transforms, '__iter__'):
```
Comment (waynegm) [inline dgbpy/torch_classes.py]: if not isinstance(transforms, list):

File: dgbpy/dgbtorch.py (hunk 1 above)
Comment (waynegm) [inline dgbpy/dgbtorch.py]: transform_list

**Review by waynegm — CHANGES_REQUESTED** [review]

File: dgbpy/torch_classes.py (hunk 2 above)
Comment (marxprop) [reply dgbpy/torch_classes.py]: I think `if not label` might not work with numpy arrays..
Also, a label image with only values of '0' might result into issues.

File: dgbpy/torch_classes.py (hunk 4 above)
Comment (marxprop) [reply dgbpy/torch_classes.py]: Thanks a lot...
I was actually making it flexible incase users insert a tuple of transforms as well.
I will keep it to just list and tuple.
Using a dictionary with this will also give issues..

File: dgbpy/dgbtorch.py (hunk 1 above)
Comment (marxprop) [reply dgbpy/dgbtorch.py]: Thanks. :innocent:

**Review by waynegm — APPROVED** [review]
